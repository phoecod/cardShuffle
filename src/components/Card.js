import React from 'react';

const Card = (props) => {
    return (
        <div className="card-container">
            <img className="card-img" src={'/imgs/deck/png/' + props.card.value + props.card.suit + '.png'} />
        </div>
        
    );
}



export default Card;