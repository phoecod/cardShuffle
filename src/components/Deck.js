import React from 'react';
import {connect} from 'react-redux';

const Deck = (props) => {
    return (
        <div className="card-container">
            {props.deckLength > 0 && <img className="card-img" src={'/imgs/deck/png/back.png'} />}
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        deckLength: state.deck.length
    }
}

export default connect(mapStateToProps,undefined)(Deck);