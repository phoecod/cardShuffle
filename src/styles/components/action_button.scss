.btn {
    border: none;
    padding: 5px 10px;
    margin: 0 10px;
    min-width: 100px;
    min-height: 40px;
    -webkit-box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
    -moz-box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
    box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
    cursor: pointer;
    font-weight: bold;
}

.btn:hover {
    opacity: 1.5;
}

.btn-container {
    display: flex;
    justify-content: center;
    margin: 15px 0;
}

.deal {
    background-color: $deal-btn;
    color: white;
}

.shuffle {
    background-color: $shuffle-btn;
    color: white;
}

.disabled {
    filter: brightness(80%);
    cursor: default;
}