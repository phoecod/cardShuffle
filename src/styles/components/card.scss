.card-container {
    width:157px;
    height:238px;
    margin: 15px 25px;
    -webkit-box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
    -moz-box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
    box-shadow: 0px 4px 6px 0px rgba(122,113,122,1);
}

.card-img {
    object-fit: contain;
    width: 100%;
    height: 100%;
}