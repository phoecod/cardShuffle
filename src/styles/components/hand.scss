.game-container {
    display: flex;
    justify-content: space-between;
    margin-bottom: 50px;
    padding-bottom: 15px;
}

