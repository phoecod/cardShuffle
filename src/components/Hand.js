import React from 'react';
import Card from './Card';
import Deck from './Deck';
import {connect} from 'react-redux';

export const Hand = (props) => {
    const deltCard = props.card;
    return (
        <div className="game-container">
              {!props.isDeckEmpty ? <Deck></Deck> : <div></div>}
            {Object.keys(deltCard).length > 0 && <Card card={deltCard} ></Card>}
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        card: state.card,
        isDeckEmpty: state.isDeckEmpty
    }
}

export default connect(mapStateToProps, undefined)(Hand);