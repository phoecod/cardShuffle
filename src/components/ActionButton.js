import React from 'react';
import {connect} from 'react-redux';
import {startDealCard, removeCard} from '../actions/card';  
import {setDeck} from '../actions/deck';  
import {setDeckState} from '../actions/deckState';  
import {removeFromDeck} from '../actions/deck';  
import {shuffle, createDeck} from '../cardActions/cardActions';

export class ActionButton extends React.Component {
    constructor (props) {
        super(props);
    }
    componentDidUpdate() {
        const deckLength = this.props.deck.length;
        if (deckLength === 0) {
            console.log("reset");
            this.props.setDeckState(true);
        }
    }

    handleShuffle = () => {
        if (this.props.isDeckEmpty) {
            this.props.removeCardFromState();
            this.props.setDeck(shuffle(createDeck()));
            this.props.setDeckState(false);
        } else {
            this.props.setDeck(shuffle(this.props.deck));
        }
    }

    handleDeal = () => {
        const deckLength = this.props.deck.length;
        const deck = this.props.deck;
        
        if (!this.props.isDeckEmpty) {
            const cardIndex = Math.floor(Math.random() * deckLength);
            this.props.startDealCard(deck[cardIndex]);
            this.props.removeFromDeck(deck[cardIndex]);
        }
    }

    render () {
        return (
            <div className="btn-container">
                <button className="btn shuffle" onClick={this.handleShuffle}>Shuffle</button>
                <button type="button" disabled={this.props.isDeckEmpty} 
                        className={this.props.isDeckEmpty ? "btn deal disabled" : "btn deal"} 
                        onClick={this.handleDeal}>Deal
                </button>
            </div>
            
        )
    }
}

const mapDispatchToProps = (dispatch) => {
	return {
        startDealCard: (card) =>  dispatch(startDealCard(card)),
        setDeck: (deck) => dispatch(setDeck(deck)),
        removeCardFromState: () => dispatch(removeCard()),
        removeFromDeck: (card) => dispatch(removeFromDeck(card)),
        setDeckState: (state) => dispatch(setDeckState(state))
	}
}

const mapStateToProps = (state) => {
    return {
        card: state.card,
        deck: state.deck,
        isDeckEmpty: state.isDeckEmpty
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ActionButton);