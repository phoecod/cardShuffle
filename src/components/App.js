import React from 'react';
import Hand from './Hand.js';
import ActionButton from '../components/ActionButton.js';
import { hot } from 'react-hot-loader'

const App = () => {
    return (
        <div className="container">
            <Hand></Hand>
            <ActionButton></ActionButton>
        </div>     
    )
}

export default hot(module)(App);