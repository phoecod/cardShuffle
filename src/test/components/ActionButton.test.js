import { shallow } from 'enzyme';
import React from 'react';
import {ActionButton} from '../../components/ActionButton';

test('should render actionButton correctly', ()=>{
    const wrapper = shallow(<ActionButton/>);
	expect(wrapper).toMatchSnapshot();
});