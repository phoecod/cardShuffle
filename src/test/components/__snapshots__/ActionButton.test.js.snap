// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render actionButton correctly 1`] = `
<div
  className="btn-container"
>
  <button
    className="btn shuffle"
    onClick={[Function]}
  >
    Shuffle
  </button>
  <button
    className="btn deal"
    onClick={[Function]}
    type="button"
  >
    Deal
  </button>
</div>
`;
